### DISCUSSION TITLE
How to implement google consent mode support in GoogleAnalytics component from `@next/third-parties/google`?

### DISCUSSION THREAD

**Discussion by aylafju** [vercel/next.js#66718] [Help]
### Summary

I need to set up the advanced google consent mode as defined [here](https://developers.google.com/tag-platform/security/guides/consent?hl=de&consentmode=advanced).

Is there a way to integrate this when using GoogleAnalytics imported from `'@next/third-parties/google'` in `next.js 14`?

### Additional information

I did this my own by implementing a component:

```js 

'use client';

import Script from 'next/script'

export default function GoogleAnalytics({gaId}){

    return (
        <>
            <Script strategy="afterInteractive" 
                src={`https://www.googletagmanager.com/gtag/js?id=${gaId}`}/>
            <Script id='google-analytics' strategy="afterInteractive"
                dangerouslySetInnerHTML={{
                __html: `
                window.dataLayer = window.dataLayer || [];
                function gtag(){dataLayer.push(arguments);}
                gtag('js', new Date());

                gtag('consent', 'default', {
                    'ad_storage': 'denied',
                    'ad_user_data': 'denied',
                    'ad_personalization': 'denied',
                    'analytics_storage': 'denied'
                });
                
                gtag('config', '${gaId}', {
                    page_path: window.location.pathname,
                });
                `,
                }}
            />
        </>
)}
```

and updating it when the user agrees or declines in the cookiebanner:

```
    useEffect(() => {
        const newValue = cookieConsent ? 'granted' : 'denied'

        window.gtag("consent", 'update', {
            'ad_user_data': newValue,
            'ad_personalization': newValue,
            'ad_storage': newValue,
            'analytics_storage': newValue
        });

        setLocalStorage("cookie_consent", cookieConsent)
    }, [cookieConsent]);

```

but using this approach I'm not able to track pageviews or custom events with `sendGAEvent`
```

**Comment by Priyanshu10230** [comment]
To **[implement Google Consent Mode](https://www.tatvic.com/blog/consent-mode-v2-implementation/?utm_campaign=forum_answer&utm_source=github&utm_medium=blog&utm_term=implementing%20google%20consent%20mode)** support in the GoogleAnalytics component from @next/third-parties/google, follow these steps:

- **Initialize Google Analytics:** Set up Google Analytics in your project using @next/third-parties/google.
- **Add Consent Mode Scripts:** Include the necessary Google Consent Mode scripts in your <head> section.
- **Configure Consent Mode:** Use the gtag function to configure Consent Mode. You need to set the default consent state for ad_storage and analytics_storage:
```
gtag('consent', 'default', {
  'ad_storage': 'denied',
  'analytics_storage': 'denied'
});
```
- **Update Consent Status:** When a user grants or denies consent via your consent management platform (CMP), update the consent state accordingly:
```
gtag('consent', 'update', {
  'ad_storage': 'granted',
  'analytics_storage': 'granted'
});
```
Implementing Google Consent Mode is crucial for complying with privacy laws and maintaining data integrity."
